resources/java/WhileAndForLoop.java

```diff
@@ -0,0 +1,12 @@
+class Loop {
+    public void loopMethod(int n) {
+        for (int i = 0; i < n; i++) {
+            System.out.println("Iteration " + (i + 1));
+        }
+        int counter = 0;
+        while (counter < n) {
+            System.out.println("Iteration " + (counter + 1));
+            counter++;
+        }
+    }
```

Comment:
According to https://docs.sonarsource.com/sonarqube/latest/user-guide/metric-definitions/, SonarQube does increase the complexity by one for "if, for, while, case, &&, ||, ?, ->". That means that in your example, it would actually increase the complexity by one, but not because of the "foreach"-call itself, but because of the lambda function with the arrow operator. 

I guess that whole discussion should be part of #41, however

Reply:
This question is not blocking the extension of more cases for control structures in this code file.